app/lib/katello/resources/candlepin/consumer.rb

```diff
@@ -240,6 +240,12 @@ def update(url, client_cert, client_key, ca_file, attributes)
             RestClient.proxy = ""
           end
 
+          def bind_entitlement(**pool)
+            JSON.parse(self['entitlements'].post(nil, params: pool))
+          ensure
+            RestClient.proxy = ""
```

Comment:
@jturel We forgot this in your PR. Would you mind opening a small PR to add it? Without this, the cdn_proxy is set forever on that thread, and subsequent calls to local candlepin try to go through the proxy.

Reply:
I don't mind but can't this be `ensure` be moved to the UpstreamCandlepinResource.resource instead of putting it in the necessary places by hand?

Reply:
Oh, I see it need needs to happen after the get/put/whatever. Hmm!

Reply:
Yeah, room for improvement here, but this is how we did it for past communication with the portal, I just moved it around ;)